test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -436,3 +436,20 @@ describe('when working with Worker threads', () => {
     strictEqual(signal, null);
   });
 });
+
+describe('Top-level await error in CommonJS modules', () => {
+  it('should throw a SyntaxError with guidance for top-level await in CommonJS', async () => {
+    const { code, stderr, stdout } = await spawnPromisified(process.execPath, [
+      '--eval',
+      'await Promise.resolve();',
+      '--input-type=commonjs',
+    ]);
+
+    match(
+      stderr,
+      /await is only valid in async functions and the top level bodies of modules/
```

Comment:
The assertion should include some of the new text you’re adding, such as “To use top-level await,” to verify that your new error message is what’s getting printed.